service/storage_service.cc

```diff
@@ -7168,5 +7170,52 @@ future<> storage_service::node_ops_abort_thread() {
     __builtin_unreachable();
 }
 
+static thread_local std::unordered_map<session_id, std::unique_ptr<session>> sessions;
+static thread_local session::list_type closing_sessions;
+
+session::guard enter_session(session_id id) {
+    if (!sessions.contains(id)) {
+        throw std::runtime_error(fmt::format("Session not found: {}", id));
+    }
+    auto guard = sessions[id]->enter();
+    slogger.debug("session {} entered", id);
+    return guard;
+}
+
+void create_session(session_id id) {
+    auto [i, created] = sessions.emplace(id, std::make_unique<session>(id));
+    if (created) {
+        slogger.debug("session {} created", id);
+    } else {
+        slogger.debug("session {} already exists", id);
+    }
+}
+
+void close_session(session_id id) noexcept {
```

Comment:
close() implies that after the call returns, we're done with the session (like close(2) or input_stream::close(). We need a name that suggests something else. Maybe start_drain_session().

Reply:
Changed to start_closing().